### PR TITLE
Smooth doors + multi-tile door code tweaks

### PR DESCRIPTION
**Pull request by TheGreatJorge**
-Changes almost all of door opening/closing animations to be smooth. Also adjusts some door sprites so they actually can be opened smoothly.
-Fixes minor problems with some door sprites. Usually only few pixels.
-Adds "o_door_open" opened state with panel open to common glass door, lift door and vault door, since these have visible panel even while opened. EDIT: They will now show up correctly when needed. Added a var used to mark doors with panels visible while open.
-Adds opening/closing animations "door_closing_stat" and "door_opening_stat" with lights off to doors that did not have it. EDIT: Should now correctly show up when the door is forced manually.
-Removes many unused duplicate dmi files and icon states in door file.
-Adds new dmi file in doors folder containing misc door overlays. Not used anywhere, purely for reference and spriting.
-Changes the multi-tile airlock to have 4 directions instead of 2. Also makes it correctly update hatches for those directions when moved or built from scratch.
EDIT:
-Adds vault assembly. It can be built like any other door, but it costs 10 sheets instead of 4 and takes double the time to make, since it has double explosion resistance and lot more health.
-Adds elevator door assembly along with icon states into the code. It is currently unused, because I have yet to figure out how to make building lift-doors work, and that will not be included in this PR.

Examples of door animations:
https://i.gyazo.com/5bfc2499d653315882af302270357477.mp4
https://i.gyazo.com/ac009387ce4af3a36057d9d108512c7e.mp4
https://i.gyazo.com/b8e3214ad54a77071e53824ec14cb5a3.mp4

Added "overlays" folder:
https://i.gyazo.com/ac3eff18c1438e2144cc8f1371d39ef4.png

### CODE REVIEW

**Comment by Lohikar** [comment]
I am still on the fence on this, the smooth animations look off in-game.

**Comment by TheGreatJorge** [comment]
Yeah, I get that. But I tried a lot of stuff and this seems to work out best. Probably not ideal, but best. If it is still not good, then there is not much I can do. I am quite fine with this being rejected at this point. But I think that deciding is not up to me, now.

**Comment by skull132** [comment]
I kinda like it. Goon has had animations this smooth for a while, and you basically just get used to it.

**Comment by TheGreatJorge** [comment]
Added some code changes and updated the description. Should be complete now and ready for review, hopefully.

File: code/game/machinery/doors/airlock.dm [hunk 1]
```diff
@@ -35,6 +35,7 @@
 	var/secured_wires = 0
 	var/datum/wires/airlock/wires = null
 	var/obj/item/device/magnetic_lock/bracer = null
+	var/panel_visible_while_open = 0
```
Comment (Lohikar) [inline code/game/machinery/doors/airlock.dm]: `FALSE`

File: code/game/machinery/doors/airlock.dm [hunk 2]
```diff
@@ -174,9 +176,10 @@
 	explosion_resistance = 20
 	opacity = 1
 	secured_wires = 1
-	assembly_type = /obj/structure/door_assembly/door_assembly_highsecurity //Until somebody makes better sprites.
+	assembly_type = /obj/structure/door_assembly/door_assembly_vault
 	hashatch = 0
 	maxhealth = 800
+	panel_visible_while_open = 1
```
Comment (Lohikar) [inline code/game/machinery/doors/airlock.dm]: `TRUE`

File: code/modules/turbolift/turbolift_door.dm [hunk 3]
```diff
@@ -6,7 +6,9 @@
 	autoclose = 0
 	glass = 1
 	icon = 'icons/obj/doors/doorlift.dmi'
+	assembly_type = /obj/structure/door_assembly/door_assembly_lift
 	hashatch = 0
+	panel_visible_while_open = 1
```
Comment (Lohikar) [inline code/modules/turbolift/turbolift_door.dm]: `TRUE`

**Comment by skull132** [comment]
:ree: Lohikar got ahead of me. Do as he said and we'll merge it, @TheGreatJorge .

**Comment by TheGreatJorge** [comment]
Should be all converted, hopefully it's alright and fine. @skull132

File: code/game/machinery/doors/multi_tile.dm [hunk 4]
```diff
@@ -7,10 +7,14 @@
...
 	. = ..()
 	SetBounds()
+	setup_hatch()
+	update_icon()
 
 /obj/machinery/door/airlock/multi_tile/Move()
 	. = ..()
 	SetBounds()
+	setup_hatch()
+	update_icon()
```
Comment (Lohikar) [inline code/game/machinery/doors/multi_tile.dm]: Not so sure on this, but these doors are uncommon enough that it won't be a problem.

**Pull request merged**